packages/core/src/components/menu/_common.scss

```diff
@@ -45,6 +45,15 @@ $dark-menu-item-color-active: $dark-minimal-button-background-color-active !defa
     color: $pt-text-color-disabled;
   }
 
+  &.pt-menu-item-single-line {
+    // Special case where text is wrapped in pt-text-overflow-ellipsis, which
+    // sets overflow: hidden. With a non-default line-height, this causes text
+    // to be clipped. Instead, we set the height directly to adhere to the
+    // grid.
+    height: $pt-grid-size * 3;
+    line-height: normal;
```

Comment:
`normal` makes the text misaligned, vertically not centered by 1px. `unset` seems to work tho, although I have no idea why?

Reply:
I can't see the misalignment on my end, so I'm assuming this is font-specific. Can you post a screenshot please?

Reply:
The misalignment appears in your own screenshot:
![](https://user-images.githubusercontent.com/205631/37629048-c4043bb4-2bdc-11e8-8622-8aea05841780.png)

Notice how "Settings..." is 1px down

Reply:
Hm, I thought that was due to Segoe UI's baseline. Here it is with `sans-serif` (Arial):
![menu5](https://user-images.githubusercontent.com/205631/37630371-84b8ef92-2be4-11e8-825a-f246475d5a81.png)

What's odd to me is that changing it from `normal` to `unset` has no effect on Firefox, but in Chrome it does actually fix the misalignment with Segoe UI.

Reply:
Another problem with `unset`: Firefox, zoom level 140%, clipping still occurs:
![menu6](https://user-images.githubusercontent.com/205631/37630964-b7647c7e-2be7-11e8-8a82-7db7f86f7fff.png)

Reply:
Had to look up the difference between `normal` and `unset`:

`normal` uses the browser default, whereas `unset` inherits the nearest parent's value. In this case it would be `<body>`, which gets its `line-height` from here:
https://github.com/palantir/blueprint/blob/b4de4e667bf31108fa8baad359aa543d9bb1e754/packages/core/src/common/_variables.scss#L30-L31

So `unset` gives us an 18px line height. Furthermore, it seems that in Chrome, odd pixel values for `line-height` seem to shift everything down by 1px, as shown here:
![2018-03-20 05-11-44 flv](https://user-images.githubusercontent.com/205631/37635681-6dca31a6-2bfd-11e8-979b-ee1f57729a65.gif)

This affects most fonts (Arial is an exception, apparently. Madness!). I'm guessing `normal` ends up producing an odd pixel value (inspector doesn't show the actual value), causing the misalignment.

Reply:
going back to the previous 18px line height try, what about:
```
.pt-menu-item {
  line-height: 18px; // math?
  padding: 6px 7px; // with the right variables math
}
.pt-menu-item .pt-icon {
  margin-top: 1px;
}
.pt-menu-item .pt-menu-item-label {
  align-self: flex-start;
}
// maybe some kind of `$vertical-padding-adjustment-size: 1px` variable to do the math above?
```
this seems to be working solidly on my end, albeit ugly. but if that solves the clipping 🤷‍♂️